### PR TITLE
Enable stylecheck linter

### PR DESCRIPTION
**Pull request by Jaanki**
Correct the variable and package names to be
compatible with stylecheck recommendations.

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -70,3 +69,12 @@ issues:
     - linters:
         - wsl
       text: "only one cuddle assignment allowed before range statement"
+
+    # These are stylecheck warnings that can be suppressed. This is temporary and will be deleted in further commits.
```
Comment (dfarrell07) [inline .golangci.yml]: This is a good strategy for breaking up very large changes like this, fixing issues in different commits per-issue-type.

In general, I found it useful to:

1. First commit to enable the comitter but add ignores for all failing rules
2. Loop over each rule, enabling it and fixing the issues it flags in a separate commit.

File: main.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func main() {
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
-	httpServer := startHttpServer()
+	httpServer := starthttpServer()
```
Comment (tpantelis) [inline main.go]: hmm... this seems odd to me that the linter would want lower case. According to [guidelines](https://github.com/golang/go/wiki/CodeReviewComments#initialisms), protocols et al should be capitalized, so `startHTTPServer`.

File: pkg/globalnet/controllers/ipam/constants.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ type Operation string
 
 const (
 	handlerResync          = time.Hour * 24
-	submarinerIpamGlobalIp = "submariner.io/globalIp"
+	submarinerIpamGlobalIP = "submariner.io/globalIP"
```
Comment (tpantelis) [inline pkg/globalnet/controllers/ipam/constants.go]: We can't change the label name w/o changing it elsewhere and causing backwards compatibility and migration issues.  We should leave "_globalIp_" as is.

File: pkg/cableengine/syncer/syncer.go [hunk 4]
```diff
@@ -70,41 +70,41 @@ func NewGatewaySyncer(engine cableengine.Engine, client v1typed.GatewayInterface
 	}
 }
 
-func (s *GatewaySyncer) Run(stopCh <-chan struct{}) {
+func (gws *GatewaySyncer) Run(stopCh <-chan struct{}) {
```
Comment (tpantelis) [inline pkg/cableengine/syncer/syncer.go]: Usually 1 or 2 letters suffices as outlined [here](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names).

```suggestion
func (gs *GatewaySyncer) Run(stopCh <-chan struct{}) {
```

File: pkg/globalnet/main.go [hunk 5]
```diff
@@ -56,7 +56,7 @@ func main() {
 	// set up signals so we handle the first shutdown signal gracefully
 	stopCh := signals.SetupSignalHandler()
 
-	httpServer := startHttpServer()
+	httpServer := starthttpServer()
```
Comment (tpantelis) [inline pkg/globalnet/main.go]:
```suggestion
	httpServer := startHTTPServer()
```

File: pkg/routeagent_driver/cniinterface/cni_iface.go [hunk 6]
```diff
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package cni_interface
+package cniinterface
```
Comment (tpantelis) [inline pkg/routeagent_driver/cniinterface/cni_iface.go]: Since _interface_ is already part of CNI:

```suggestion
package cni
```

File: pkg/routeagent_driver/constants/constants.go [hunk 7]
```diff
@@ -36,10 +36,10 @@ const (
 	// part of route-agent and this will subsequently be used in globalnet controller for
 	// supporting connectivity from HostNetwork to remoteClusters.
 	// [#] interface on the node that has an IPAddress from the clusterCIDR
-	CniInterfaceIP = "submariner.io/cniIfaceIp"
+	CniInterfaceIP = "submariner.io/cniIfaceIP"
 
 	// TODO (revisit): Unify this constant across different components
-	SmGlobalIP = "submariner.io/globalIp"
+	SmGlobalIP = "submariner.io/globalIP"
```
Comment (tpantelis) [inline pkg/routeagent_driver/constants/constants.go]: As above we shouldn't change label names.

File: pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go [hunk 8]
```diff
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package kubeproxy_iptables
+package kubeproxyiptables
```
Comment (tpantelis) [inline pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go]: I think the '_' makes it more readable.  In lieu of that, how about we just make it _kubeproxy_?

File: pkg/util/clusterfiles/cluster_files.go [hunk 9]
```diff
@@ -13,7 +13,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package cluster_files
+package clusterfiles
```
Comment (tpantelis) [inline pkg/util/clusterfiles/cluster_files.go:16]: How about shorten it to just _files_?

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/globalnet/controllers/ipam/constants.go (hunk 3 above)
Comment (mangelajo) [reply pkg/globalnet/controllers/ipam/constants.go]: Correct, the submariner.io/globalIp part needs to stay as it 's (the constant name can change, but not the content).

File: main.go (hunk 2 above)
Comment (mangelajo) [reply main.go]: Correct

File: pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go (hunk 8 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go]: agreed about readability but I don't see a good match that works with the "kubeproxyiptables" directory name? can they be different, if they can be different ... may be

File: pkg/util/clusterfiles/cluster_files.go (hunk 9 above)
Comment (mangelajo) [reply pkg/util/clusterfiles/cluster_files.go:16]: I think also clusterfiles would be ok, not as long as "kubeproxyiptables" ':D this starts to sound like german

**Review by mangelajo — CHANGES_REQUESTED** [review]
I agree with the first two or three by tom

File: pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go (hunk 8 above)
Comment (Jaanki) [reply pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go]: directory name and package name can be different but its a convention that they should be the same (not necessarily though) so I changed the directory name.

Comment (Jaanki) [reply pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go]: How about kpiptables?

Comment (tpantelis) [reply pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go]: @mangelajo How about just shorten to _kubeproxy_? But if the _iptables_ part is important, ie there could be another kubeproxy implementation,  then how about a subdirectory _kubeproxy/iptables_?

Re: '_' in general in package names, I'm OK with it although we should still strive for brevity to avoid the need - if others agree, perhaps we can disable this check/failure?

File: .golangci.yml [hunk 10]
```diff
@@ -70,3 +69,20 @@ issues:
...
+    # Allow dot imports for `fake` directory
+    - linters:
+        - stylecheck
+      text: "ST1001: "
+      path: pkg/cable/fake
+
+    - linters:
+        - stylecheck
+      text: "ST1001: "
+      path: pkg/cableengine/fake
```
Comment (tpantelis) [inline .golangci.yml]: We can handle all _fake_ dirs with one rule. Also need to exclude _test/e2e_.

```suggestion
    # Allow dot imports for gingko and gomega in test-related directories
    - linters:
        - stylecheck
      text: "ST1001: "
      path: /fake

    - linters:
        - stylecheck
      text: "ST1001: "
      path: test/e2e
```

File: .golangci.yml [hunk 11]
```diff
@@ -70,3 +69,20 @@ issues:
     - linters:
         - wsl
       text: "only one cuddle assignment allowed before range statement"
+
+    # Ignore new line warning at L30. New line is added to reduce line length according to lll
+    - linters:
+        - stylecheck
+      text: "ST1005: "
+      path: pkg/networkplugin-syncer/handlers/ovn/handler.go
```
Comment (tpantelis) [inline .golangci.yml]: The error seems like a false-positive bug with the linter. We should ignore this one line in code via:

`//nolint:stylecheck This seems like a false-positive bug with the linter - the actual string has no newline`

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 11 above)
Comment (Jaanki) [reply .golangci.yml]: ahh yes...thats a good alternative. Forgot this one

File: pkg/networkplugin-syncer/handlers/ovn/handler.go [hunk 12]
```diff
@@ -27,6 +27,7 @@ import (
 	"github.com/submariner-io/submariner/pkg/networkplugin-syncer/handlers/ovn/nbctl"
 )
 
+//nolint:stylecheck // Limit line length as per lll
```
Comment (tpantelis) [inline pkg/networkplugin-syncer/handlers/ovn/handler.go]: The "_ST1005: error strings should not end with punctuation or a newline_" error is actually due to the ending period and not line-length. So we just need to remove the period 😄

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go (hunk 8 above)
Comment (mangelajo) [reply pkg/routeagent_driver/handlers/kubeproxyiptables/cleanup_handling.go]: Yeah, that makes sense we could have kubeproxy now, and if we ever support the other implementation use folders.

File: .golangci.yml [hunk 13]
```diff
@@ -70,3 +69,14 @@ issues:
...
+    # Allow dot imports for `fake` directory
+    - linters:
+        - stylecheck
+      text: "ST1001: "
+      path: /fake
+
+    - linters:
+        - stylecheck
+      text: "ST1001: "
+      path: test/e2e
```
Comment (mangelajo) [inline .golangci.yml]: This needs a return at the end of the line.

**Comment by mangelajo** [comment]
No more comments on my side, just what @tpantelis said & agreeing with "kubeproxy" instead of kubeproxyiptables, but that's a nit :-P :D

**Review by mangelajo — CHANGES_REQUESTED** [review]

File: pkg/networkplugin-syncer/handlers/ovn/handler.go [hunk 14]
```diff
@@ -27,7 +27,8 @@ import (
 	"github.com/submariner-io/submariner/pkg/networkplugin-syncer/handlers/ovn/nbctl"
 )
 
-var WaitingForLocalEndpoint = errors.New("Waiting for the local endpoint details before we can " +
+//nolint:stylecheck // Limit line length as per lll
+var ErrWaitingForLocalEndpoint = errors.New("Waiting for the local endpoint details before we can " +
 	"setup any remote endpoint related information, this will be retried.")
```
Comment (skitt) [inline pkg/networkplugin-syncer/handlers/ovn/handler.go]:
```suggestion
var ErrWaitingForLocalEndpoint = errors.New("waiting for the local endpoint details before we can " +
	"setup any remote endpoint related information, this will be retried")
```

File: pkg/networkplugin-syncer/handlers/ovn/nbctl/nbctl.go [hunk 15]
```diff
@@ -103,9 +103,9 @@ func (n *NbCtl) GetGatewayChassis(lrp, chassis string) (string, error) {
 	return output, err
 }
 
-func (n *NbCtl) LrSetLoadBalancers(lrId string, lbIDs []string) error {
+func (n *NbCtl) LrSetLoadBalancers(lrID string, lbIDs []string) error {
```
Comment (skitt) [inline pkg/networkplugin-syncer/handlers/ovn/nbctl/nbctl.go:106]: Although the linter doesn’t flag this, `Lr` should be `LR` throughout since it’s an acronym. (But we can ignore this for now.)

File: pkg/networkplugin-syncer/handlers/ovn/services.go [hunk 16]
```diff
@@ -29,23 +29,23 @@ func (ovn *SyncHandler) ensureServiceLoadBalancersFrom(logicalSwitch string) err
 		return err
 	}
 
-	submarinerRouterId, err := ovn.getSubmarinerRouterId()
+	submarinerRouterID, err := ovn.getSubmarinerRouterID()
```
Comment (skitt) [inline pkg/networkplugin-syncer/handlers/ovn/services.go:32]: “Id” is an abbreviation, not an acronym, but this is a bug in the linter...

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/networkplugin-syncer/handlers/ovn/services.go (hunk 16 above)
Comment (skitt) [reply pkg/networkplugin-syncer/handlers/ovn/services.go:32]: And the rest of the world disagrees with me, so “ID” it is.

File: pkg/networkplugin-syncer/handlers/ovn/nbctl/nbctl.go (hunk 15 above)
Comment (Jaanki) [reply pkg/networkplugin-syncer/handlers/ovn/nbctl/nbctl.go:106]: Ignoring for now.

File: .golangci.yml (hunk 13 above)
Comment (dfarrell07) [inline .golangci.yml]: Can we have this comment explain both rules?

File: pkg/globalnet/controllers/ipam/ipam.go [hunk 17]
```diff
@@ -364,19 +364,19 @@ func (i *Controller) handleUpdateNode(old, newObj interface{}) {
 		return
 	}
 
-	oldCniIfaceIpOnNode := old.(*k8sv1.Node).GetAnnotations()[constants.CniInterfaceIP]
-	newCniIfaceIpOnNode := newObj.(*k8sv1.Node).GetAnnotations()[constants.CniInterfaceIP]
-	if oldCniIfaceIpOnNode == "" && newCniIfaceIpOnNode != "" {
-		klog.V(log.DEBUG).Infof("In handleUpdateNode, node %q is now annotated with cniIfaceIP, enqueing", newObj.(*k8sv1.Node).Name)
+	oldCniIfaceIPOnNode := old.(*k8sv1.Node).GetAnnotations()[constants.CniInterfaceIP]
```
Comment (dfarrell07) [inline pkg/globalnet/controllers/ipam/ipam.go:367]: I guess the linter doesn't flag "Cni"? This is a bit of a strange var name, it expands to include "interface interface". I guess this is out of scope for this PR.

File: pkg/globalnet/controllers/ipam/ipam.go [hunk 18]
```diff
@@ -442,31 +442,31 @@ func (i *Controller) handleRemovedPod(obj interface{}) {
 	}
 
 	if !i.excludeNamespaces[pod.Namespace] {
-		globalIp := pod.Annotations[SubmarinerIpamGlobalIp]
-		if globalIp != "" && pod.Status.PodIP != "" {
+		globalIP := pod.Annotations[SubmarinerIpamGlobalIP]
```
Comment (dfarrell07) [inline pkg/globalnet/controllers/ipam/ipam.go:445]: I guess the linter doesn't flag "Ipam"

File: pkg/routeagent_driver/cni/cni_iface.go [hunk 19]
```diff
@@ -79,7 +79,7 @@ func ConfigureRpFilter(iface string) error {
 	// #nosec G306
 	err := ioutil.WriteFile("/proc/sys/net/ipv4/conf/"+iface+"/rp_filter", []byte("2"), 0644)
 	if err != nil {
-		return fmt.Errorf("unable to update rp_filter for cni_interface %q, err: %s", iface, err)
+		return fmt.Errorf("unable to update rp_filter for cni interface %q, err: %s", iface, err)
```
Comment (dfarrell07) [inline pkg/routeagent_driver/cni/cni_iface.go:82]: CNI?

File: pkg/globalnet/controllers/ipam/ipam.go (hunk 17 above)
Comment (Jaanki) [reply pkg/globalnet/controllers/ipam/ipam.go:367]: I have only changed the names that the linter flagged.

File: pkg/globalnet/controllers/ipam/ipam.go (hunk 18 above)
Comment (Jaanki) [reply pkg/globalnet/controllers/ipam/ipam.go:445]: Ditto

File: pkg/routeagent_driver/cni/cni_iface.go (hunk 19 above)
Comment (Jaanki) [reply pkg/routeagent_driver/cni/cni_iface.go:82]: This applies to all `cni`s throughtout the codebase and its out of the scope of this patch.

File: pkg/globalnet/controllers/ipam/ipam.go (hunk 17 above)
Comment (tpantelis) [reply pkg/globalnet/controllers/ipam/ipam.go:367]: CNI isn't a well-known acronym outside k8s, like ID and IP, so the linter doesn't know about it.

Comment (skitt) [reply pkg/globalnet/controllers/ipam/ipam.go:367]: We can add our own configuration with other acronyms we want to watch for...

File: pkg/routeagent_driver/cni/cni_iface.go (hunk 19 above)
Comment (Jaanki) [reply pkg/routeagent_driver/cni/cni_iface.go:82]: This applies to all `CNI Interfaces` throughtout the code and is out of the scope of this PR.

File: pkg/routeagent_driver/constants/constants.go [hunk 20]
```diff
@@ -36,7 +36,7 @@ const (
 	// part of route-agent and this will subsequently be used in globalnet controller for
 	// supporting connectivity from HostNetwork to remoteClusters.
 	// [#] interface on the node that has an IPAddress from the clusterCIDR
-	CniInterfaceIP = "submariner.io/cniIfaceIp"
+	CniInterfaceIP = "submariner.io/cniIfaceIP"
```
Comment (tpantelis) [inline pkg/routeagent_driver/constants/constants.go]: Changing the label name may have other effects and would be outside the scope of this PR.

```suggestion
	CniInterfaceIP = "submariner.io/cniIfaceIp"
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/routeagent_driver/cni/cni_iface.go (hunk 19 above)
Comment (dfarrell07) [reply pkg/routeagent_driver/cni/cni_iface.go:82]: Kinda, I mean we are adding this comment and introducing a new "cni" that should be "CNI". As @skitt suggested, probably best to raise an issue to track adding additional config for custom acronyms we want to watch for.

Comment (tpantelis) [reply pkg/routeagent_driver/cni/cni_iface.go:82]: It's referred to in several different ways in messages in this file, ie _CNI Interface_, _cni_interface_, _cniInterface_, _cniIface_. We should be consistent - IMO _CNI_ as @dfarrell07 suggest. However I do agree it's a bit out of scope for this PR - this PR is already quite large and refactors quite a bit of files so it would be nice to get this merged ASAP. I have work to do in several of the files but have been holding off to avoid significant conflict resolution.

Comment (tpantelis) [reply pkg/routeagent_driver/cni/cni_iface.go:82]: @Jaanki Can  you address this in another PR?

**Review by tpantelis — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Review by mangelajo — APPROVED** [review]

File: pkg/cable/cleanup/xfrm_cleanup_handler.go [hunk 21]
```diff
@@ -51,7 +51,7 @@ func (xc *XFRMCleanupHandler) NonGatewayCleanup() error {
 		klog.V(log.DEBUG).Infof("Deleting XFRM policy %s", currentXfrmPolicyList[i])
 
 		if err = netlink.XfrmPolicyDel(&currentXfrmPolicyList[i]); err != nil {
-			return fmt.Errorf("Error Deleting XFRM policy %s: %v", currentXfrmPolicyList[i], err)
+			return fmt.Errorf("error Deleting XFRM policy %s: %v", currentXfrmPolicyList[i], err)
```
Comment (mangelajo) [inline pkg/cable/cleanup/xfrm_cleanup_handler.go:54]:
```suggestion
			return fmt.Errorf("error deleting XFRM policy %s: %v", currentXfrmPolicyList[i], err)
```

**Review by skitt — APPROVED** [review]
